consensus/src/metrics_safety_rules.rs

```diff
@@ -35,16 +38,34 @@ impl MetricsSafetyRules {
     pub fn perform_initialize(&mut self) -> Result<(), Error> {
         let consensus_state = self.consensus_state()?;
         let sr_waypoint = consensus_state.waypoint();
-        let proofs = self
-            .storage
-            .retrieve_epoch_change_proof(sr_waypoint.version())
-            .map_err(|e| {
-                Error::InternalError(format!(
-                    "Unable to retrieve Waypoint state from storage, encountered Error:{}",
-                    e
-                ))
-            })?;
-        self.initialize(&proofs)
+        loop {
+            let proofs = self
+                .storage
+                .retrieve_epoch_change_proof(sr_waypoint.version())
+                .map_err(|e| {
+                    Error::InternalError(format!(
+                        "Unable to retrieve Waypoint state from storage, encountered Error:{}",
+                        e
+                    ))
+                })?;
+            let result = self.initialize(&proofs);
+            // If the provided epoch is still far behind the current epoch,
```

Comment:
nit: I'm not sure we need this long of a comment, maybe something more concise above the loop? e.g.,

`We keep initializing safety rules as long as the waypoint continues to increase. This is due to limits in the number of epoch change proofs that storage can provide`.

Reply:
Made the comment more concise. Used the suggested comment.